utils/auth_utils.py

```diff
@@ -90,37 +96,44 @@ def entity_data(token_data: dict) -> str:
     entity_id = token_data.get('entity_id_data', None)
 
     if wrapper and entity_class and endpoint and entity_id:
-        xml = wrapper.read_object(endpoint=endpoint, obj={"id":entity_id})[0]
-        
-        if type(xml) == type(None):
-            pass # TODO
-        elif xml.type != "Quality Control":
-            pass # TODO
-        elif xml.status == "finished":
-            pass # TODO
-
-        bfabric_positions = []
-        bfabric_ids = []
-
-        for x in range(len(xml.sample)):  #for all samples on plate
-            bfabric_positions.append(xml.sample[x]._gridposition)  # get sample plate position
-            bfabric_ids.append(xml.sample[x]._id) # get sample id
-
+        entity_data_dict = wrapper.read(endpoint=endpoint, obj={"id": entity_id})[0]
+
+        if entity_data_dict:
+            if entity_data_dict.get("type") != "Quality Control":
+                pass  # TODO
+            elif entity_data_dict.get("status") == "finished":
+                pass  # TODO
+
+            bfabric_positions = []
+            bfabric_ids = []
+            samples = entity_data_dict.get("sample", [])
+            
+            for sample in samples:
+                bfabric_positions.append(sample.get("_gridposition"))
+                bfabric_ids.append(sample.get("_id"))
```

Comment:
Somehow I believe this line is not actually adding ids to the list (see agenda document)